### PR TITLE
catching promise - legacy provider error

### PR DESCRIPTION
**Pull request by luu-alex**
## Description

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](https://github.com/web3/web3.js/blob/4.x/.github/CONTRIBUTING.md).

## Type of change

#7035

Wallets such as phantom is erroring out when rejecting to sign/send a transaction when testing with web3modal in laboratory
This is due to phantom being considered as a legacy provider to web3 and throws when trying to process jsonrpcresponses.
```
// needs to be wrapped because when phantom wallet throws an erorr, no rpc code is given, so we need to catch it 
const processedError = this._processJsonRpcResponse(
						payload,
						error as JsonRpcResponse<ResponseType, unknown>,
						{ legacy: true, error: true }
					);
```

This PR catches the error and doesnt crash the app when the user rejects signing a transaction

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: packages/web3/test/integration/external-providers/helper.ts [hunk 1]
```diff
@@ -44,5 +46,31 @@ export async function performBasicRpcCalls(provider: SupportedProviders) {
 	expect(typeof blockNumber1).toBe('bigint');
 
 	// After sending a transaction, the blocknumber is supposed to be greater than or equal the block number before sending the transaction
-	expect(blockNumber1).toBeGreaterThanOrEqual(blockNumber0);
+	expect(blockNumber1).toBeGreaterThanOrEqual(blockNumber0);
```
Comment (Muhammad-Altabba) [inline packages/web3/test/integration/external-providers/helper.ts]: Could you please ensure prettier is working well on your machine? Things link this should be fixed by it. 
```suggestion
	expect(blockNumber1).toBeGreaterThanOrEqual(blockNumber0);
```

File: packages/web3-eth/src/rpc_method_wrappers.ts [hunk 2]
```diff
@@ -581,7 +581,7 @@ export function sendTransaction<
 							transaction,
 							transactionFormatted,
 						});
-
+
```
Comment (Muhammad-Altabba) [inline packages/web3-eth/src/rpc_method_wrappers.ts]: Could you please ensure prettier is working well on your machine? Things like this should be fixed by it. 

```suggestion

```

File: packages/web3-core/src/web3_request_manager.ts [hunk 3]
```diff
@@ -241,14 +241,14 @@ export class Web3RequestManager<
 						payload,
 						error as JsonRpcResponse<ResponseType, unknown>,
 						{ legacy: true, error: true },
-					),
+					)
```
Comment (Muhammad-Altabba) [inline packages/web3-core/src/web3_request_manager.ts]: Could you please ensure prettier is working well on your machine?
```suggestion
					),
```

File: packages/web3-core/src/web3_request_manager.ts [hunk 4]
```diff
@@ -241,14 +241,14 @@ export class Web3RequestManager<
...
 				);
 		}
 
 		// TODO: This could be deprecated and removed.
 		if (isLegacyRequestProvider(provider)) {
 			return new Promise<JsonRpcResponse<ResponseType>>((resolve, reject) => {
-				const rejectWithError = (err: unknown) =>
+				const rejectWithError = (err: unknown) =>{
```
Comment (Muhammad-Altabba) [inline packages/web3-core/src/web3_request_manager.ts]: Could you please ensure prettier is working well on your machine?

File: packages/web3/test/integration/external-providers/helper.ts [hunk 5]
```diff
@@ -44,5 +46,31 @@ export async function performBasicRpcCalls(provider: SupportedProviders) {
...
+	const accounts = await web3.eth.getAccounts();
+
+	const sendOptions = { from: accounts[0], gas: '1000000' };
+
+
+	contract = await contract.deploy(deployOptions).send(sendOptions);
+
+	await contract.methods.reverts().send({ from: accounts[0] })
+	
+}
```
Comment (Muhammad-Altabba) [inline packages/web3/test/integration/external-providers/helper.ts]: Could you please ensure prettier is working well on your machine?

```suggestion
}

```

File: packages/web3-core/test/unit/web3_request_manager.test.ts [hunk 6]
```diff
@@ -280,6 +280,21 @@ describe('Web3RequestManager', () => {
 			expect(myProvider.request).toHaveBeenCalledTimes(1);
 			expect(await pr).toBe('test');
 		});
+
+		it('promise of legacy provider should be error', async () => {
```
Comment (Muhammad-Altabba) [inline packages/web3-core/test/unit/web3_request_manager.test.ts]: Sorry, cannot see something related to testing promise with `error` here.

File: packages/web3-core/src/web3_request_manager.ts [hunk 7]
```diff
@@ -288,7 +289,20 @@ export class Web3RequestManager<
...
-					responsePromise.then(resolveWithResponse).catch(rejectWithError);
+					responsePromise.then(resolveWithResponse).catch(error => {
+							try {
+								// Attempt to process the error response
+								const processedError = this._processJsonRpcResponse(
+									payload,
+									error as JsonRpcResponse<ResponseType, unknown>,
+									{ legacy: true, error: true }
+								);
+								reject(processedError);
```
Comment (Muhammad-Altabba) [inline packages/web3-core/src/web3_request_manager.ts]: According to codevov, this line is not checked at unit testing.

File: packages/web3/test/integration/external-providers/helper.ts (hunk 1 above)
Comment (luu-alex) [reply packages/web3/test/integration/external-providers/helper.ts]: thank you, updated

File: packages/web3-core/test/unit/web3_request_manager.test.ts (hunk 6 above)
Comment (luu-alex) [reply packages/web3-core/test/unit/web3_request_manager.test.ts]: this was supposed to be removed, thanks

File: packages/web3-core/src/web3_request_manager.ts (hunk 7 above)
Comment (luu-alex) [reply packages/web3-core/src/web3_request_manager.ts]: covered

**Comment by luu-alex** [comment]
unrelated to PR but formatted [send_transaction.test.ts](https://github.com/web3/web3.js/pull/7022/files#diff-9a461a72ebf3e63d1ac2948efc345e1fef7a1b14533e2e6edf66fcd75d024970)

**Review by jdevcs — APPROVED** [review]

**Review by avkos — APPROVED** [review]

**Pull request merged**